cli.js

```diff
@@ -78,8 +81,34 @@ Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
         describe: 'Use directory dir. "create" will create this directory if needed.',
         type: 'string',
       })
-      .strict();
+      .strict()
+      .example(
+` // Create a function from python3 template
+  bn create python3 ninja
+
+  // Create a function from node8 template with concurrent execution model
+  bn create node8 pirate --config.executionModel=concurrent
+
+  // Create a function from python2 template with exclusive execution model and FOO env
+  bn create python2 hello --config.executionModel=exclusive --config.env.FOO=bar
+`);
   }, async (argv) => {
+    // validating here and not in coerce to keep output consistent
+    if (argv.config !== undefined) {
+      const configType = typeof argv.config;
```

Comment:
1. Not sure what you're protecting against here. No `type` is provided for the `config` option in `create`. This means that everything will come in as either an `object` or `boolean` (in the case that only a flag is passed `bn create node8 foo --config`). Unless I'm missing a parsing mechanism, I don't understand how it could ever be an array. 

2. I'm not 100% sure this is the correct way to check if something is an object. There are a lot of differing opinions [here](https://stackoverflow.com/questions/8511281/check-if-a-value-is-an-object-in-javascript) but it seems like the method you used might not cover `null`. Again, I don't think it matters because the only types that can reach that line of execution are `boolean` and `object`.

Reply:
1. 1. as the tests verify, the code is protecting against `--config=bad` which will turn `argv.config` a string for example (so, it can be more than an object or a boolean)
   2. added to the test the case of `--config=aaa --config=bbb` which generates an array.
2. since yargs does not generate null values, we're safe.

Reply:
1. Didn't think of it, 100% agree
2. A comment explaining the rationale for the checks might be nice.